adapters/repos/db/vector/hnsw/search.go

```diff
@@ -557,22 +557,33 @@ func (h *hnsw) knnSearchByVector(searchVec []float32, k int,
 	}
 
 	if h.shouldRescore() {
-		ids := make([]uint64, res.Len())
-		i := len(ids) - 1
-		for res.Len() > 0 {
-			res := res.Pop()
-			ids[i] = res.ID
-			i--
+		ids := make([]uint64, 0, res.Len())
+		for _, item := range res.Items() {
+			ids = append(ids, item.ID)
```

Comment:
Since we are preallocating `ids` with the correct length, it is more efficient to not append to the list, as each append is another allocation. Instead you can populate each index of the preallocated `ids` slice in the loop

Reply:
I believe append in this case will only change len of the slice without allocating new memory.

Reply:
I also think is should only change the size since we reserve the right capacity initially but will not hurt changing it as @parkerduckworth suggested. At the very least we will set length only once.